### PR TITLE
CI: Create codeql.yml

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

I used github's tool to configure its native vulnerability checker. It checks main and release branches after a commit is merged and on a cron schedule.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", "v[0-9].[0-9].x", "v[0-9].[0-9][0-9].x", "v[0-9].x" ]
```
Comment (SweeXordious) [inline .github/workflows/codeql.yml:16]: Needs a bit of yaml formatting to match our linter

**Review by SweeXordious — COMMENTED** [review]
:tada: I will do the same for orchestrator-relayer

**Comment by SweeXordious** [comment]
maybe also enable: https://github.com/celestiaorg/celestia-app/security/dependabot

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+jobs:
+  analyze:
+    name: Analyze
+    # Runner size impacts CodeQL analysis time. To learn more, please see:
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners
+    # Consider using larger runners for possible analysis time improvements.
+    runs-on: 'ubuntu-latest'
+    timeout-minutes: ${{ (matrix.language == 'swift' && 120) || 360 }}
```
Comment (rootulp) [inline .github/workflows/codeql.yml:29]: [no change needed] I think this could be simplified

```suggestion
    timeout-minutes: 360
```

**Review by rootulp — APPROVED** [review]

**Review by rootulp — APPROVED** [review]
Still LGTM

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**